libs/@local/hash-subgraph/src/types/shared/branded.ts

```diff
@@ -61,10 +61,17 @@ export const isEntityId = (entityId: string): entityId is EntityId => {
   );
 };
 
-export const entityIdFromOwnedByIdAndEntityUuid = (
+export const entityIdFromComponents = (
   ownedById: OwnedById,
   entityUuid: EntityUuid,
+  draftId?: DraftId,
 ): EntityId => {
+  const base = `${ownedById}${ENTITY_ID_DELIMITER}${entityUuid}`;
+
+  if (draftId) {
+    return `${base}${ENTITY_ID_DELIMITER}${draftId}` as EntityId;
+  }
+
   return `${ownedById}${ENTITY_ID_DELIMITER}${entityUuid}` as EntityId;
```

Comment:
We could just re-use `base` here
```suggestion
  return base as EntityId;
```

Reply:
https://github.com/hashintel/hash/pull/4221/commits/36c5aa74b9c28b8446a2f310a6a6edc603828326